pkg/controllers/deprovisioning/types.go

```diff
@@ -55,25 +54,16 @@ func (r Result) String() string {
 }
 
 type Deprovisioner interface {
-	// ShouldDeprovision is a predicate used to filter deprovisionable nodes
 	ShouldDeprovision(context.Context, *state.Node, *v1alpha5.Provisioner, []*v1.Pod) bool
-	// sortCandidates orders deprovisionable nodes by the deprovisioner's pre-determined priority
-	SortCandidates([]CandidateNode) []CandidateNode
-	// computeCommand generates a deprovisioning command given deprovisionable nodes
-	ComputeCommand(context.Context, int, ...CandidateNode) (Command, error)
-	// validateCommand validates a command for a deprovisioner
-	ValidateCommand(context.Context, []CandidateNode, Command) (bool, error)
-	// TTL returns the time to wait for a deprovisioner's validation
-	TTL() time.Duration
-	// String is the String representation of the deprovisioner
+	ComputeCommand(context.Context, ...CandidateNode) (Command, error)
 	String() string
 }
 
 type action byte
 
 const (
```

Comment:
It's starting to seem like the actions and the results are converging on each other, is it possible to collapase these into a single concept?